services/housekeeper/housekeeper.go

```diff
@@ -95,18 +98,75 @@ func (hk *Housekeeper) Start() (err error) {
 	}()
 
 	// Process the current slot
-	headSlot := syncStatus.HeadSlot
+	headSlot := bestSyncStatus.HeadSlot
 	hk.processNewSlot(headSlot)
 
 	// Start regular slot updates
 	c := make(chan uint64)
-	go hk.beaconClient.SubscribeToHeadEvents(c)
+	for _, client := range hk.beaconClients {
+		go client.SubscribeToHeadEvents(c)
+	}
 	for {
 		headSlot := <-c
 		hk.processNewSlot(headSlot)
 	}
 }
 
+func (hk *Housekeeper) getBestSyncStatus() (*beaconclient.SyncStatusPayloadData, error) {
```

Comment:
we have the same functionality in api too. should we split that out into the common package?

Reply:
sounds good